mm2src/coins/lightning.rs

```diff
@@ -844,82 +1240,167 @@ pub async fn send_payment(ctx: MmArc, req: SendPaymentReq) -> SendPaymentResult<
         MmCoinEnum::LightningCoin(c) => c,
         _ => return MmError::err(SendPaymentError::UnsupportedCoin(coin.ticker().to_string())),
     };
-    let nodes_addresses = ln_coin.nodes_addresses.lock().clone();
-    for (node_pubkey, node_addr) in nodes_addresses {
+    let open_channels_nodes = ln_coin.open_channels_nodes.lock().clone();
+    for (node_pubkey, node_addr) in open_channels_nodes {
         connect_to_node(node_pubkey, node_addr, ln_coin.peer_manager.clone())
             .await
             .error_log_with_msg(&format!(
                 "Channel with node: {} can't be used to route this payment due to connection error.",
                 node_pubkey
             ));
     }
-    let (payment_hash, payment_info) = match req.payment {
+    let payment_info = match req.payment {
         Payment::Invoice { invoice } => ln_coin.pay_invoice(invoice.into())?,
         Payment::Keysend {
             destination,
             amount_in_msat,
             expiry,
         } => ln_coin.keysend(destination.into(), amount_in_msat, expiry)?,
     };
-    let mut outbound_payments = ln_coin.outbound_payments.lock();
-    outbound_payments.insert(payment_hash, payment_info);
+    ln_coin
+        .persister
+        .add_or_update_payment_in_sql(payment_info.clone())
+        .await?;
     Ok(SendPaymentResponse {
-        payment_hash: payment_hash.0.into(),
+        payment_hash: payment_info.payment_hash.0.into(),
     })
 }
 
+#[derive(Deserialize)]
+pub struct PaymentsFilterForRPC {
+    pub payment_type: Option<PaymentTypeForRPC>,
+    pub description: Option<String>,
+    pub status: Option<HTLCStatus>,
+    pub from_amount_msat: Option<u64>,
+    pub to_amount_msat: Option<u64>,
+    pub from_fee_paid_msat: Option<u64>,
+    pub to_fee_paid_msat: Option<u64>,
+    pub from_timestamp: Option<u64>,
+    pub to_timestamp: Option<u64>,
+}
+
+impl From<PaymentsFilterForRPC> for PaymentsFilter {
+    fn from(filter: PaymentsFilterForRPC) -> Self {
+        PaymentsFilter {
+            payment_type: filter.payment_type.map(From::from),
+            description: filter.description,
+            status: filter.status,
+            from_amount_msat: filter.from_amount_msat,
+            to_amount_msat: filter.to_amount_msat,
+            from_fee_paid_msat: filter.from_fee_paid_msat,
+            to_fee_paid_msat: filter.to_fee_paid_msat,
+            from_timestamp: filter.from_timestamp,
+            to_timestamp: filter.to_timestamp,
+        }
+    }
+}
+
 #[derive(Deserialize)]
 pub struct ListPaymentsReq {
     pub coin: String,
+    pub filter: Option<PaymentsFilterForRPC>,
+    #[serde(default = "ten")]
+    limit: usize,
+    #[serde(default)]
+    paging_options: PagingOptionsEnum<H256Json>,
+}
+
+#[derive(Deserialize, Serialize)]
+#[serde(tag = "type")]
+pub enum PaymentTypeForRPC {
```

Comment:
> I think it's fare for public crates. At the moment, our crates are closely related to each other. I personally prefer to avoid using `_` in pattern matching to catch logic errors at the compile time. It happens that external crates must know about newly added enum variants, new fields etc.

That's also right. If we are sure that nothing is going to be published into crates.io registry, then yes we don't need to use this.